### PR TITLE
Fix WebGL2RenderingContext broken links

### PR DESCRIPTION
**Pull request by tsnee**
webgl2renderingcontext/index.md tries to link to the methods `WebGL2RenderingContext.bufferData` and `WebGL2RenderingContext.bufferSubData`, but those two methods are actually defined in `WebGLRenderingContext` (note the lack of a `2`).

### Description

Fix two broken links.

### Motivation

By linking directly to the correct file, it allows readers to skip a Google search.

### CODE REVIEW

**Comment by wbamberg** [comment]
Thank you, but as far as I can tell this isn't the right fix. `WebGLRenderingContext` and `WebGL2RenderingContext` are two distinct interfaces, and this page lists members of `WebGL2RenderingContext`. So it's not right to link to members of `WebGLRenderingContext` from it.

It looks like [both interfaces define these methods](https://registry.khronos.org/webgl/specs/latest/2.0/#3.7), so a fix would be to write pages for the WebGL2 versions of them.

**Comment by tsnee** [comment]
> Thank you, but as far as I can tell this isn't the right fix. `WebGLRenderingContext` and `WebGL2RenderingContext` are two distinct interfaces, and this page lists members of `WebGL2RenderingContext`. So it's not right to link to members of `WebGLRenderingContext` from it.

`WebGL2RenderingContext` is a subclass of `WebGLRenderingContext`. Maybe that is why the original documentation authors decided to document overloaded methods like these two in the same place, namely, under `WebGLRenderingContext`. If you look at https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/bufferData and https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/bufferSubData you will see what I mean - both WebGL 1 and WebGL 2 usages are defined.

> It looks like [both interfaces define these methods](https://registry.khronos.org/webgl/specs/latest/2.0/#3.7), so a fix would be to write pages for the WebGL2 versions of them.

That would be less confusing. However it does not seem to be what the original documentation authors intended. That said, maybe we should take this opportunity to make a bigger change and split the documents into WebGL 1 and WebGL 2 versions. I would be open to that. What do you think?

**Comment by wbamberg** [comment]
> > Thank you, but as far as I can tell this isn't the right fix. `WebGLRenderingContext` and `WebGL2RenderingContext` are two distinct interfaces, and this page lists members of `WebGL2RenderingContext`. So it's not right to link to members of `WebGLRenderingContext` from it.
> 
> `WebGL2RenderingContext` is a subclass of `WebGLRenderingContext`. Maybe that is why the original documentation authors decided to document overloaded methods like these two in the same place, namely, under `WebGLRenderingContext`. If you look at https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/bufferData and https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/bufferSubData you will see what I mean - both WebGL 1 and WebGL 2 usages are defined.
> 
> > It looks like [both interfaces define these methods](https://registry.khronos.org/webgl/specs/latest/2.0/#3.7), so a fix would be to write pages for the WebGL2 versions of them.
> 
> That would be less confusing. However it does not seem to be what the original documentation authors intended. That said, maybe we should take this opportunity to make a bigger change and split the documents into WebGL 1 and WebGL 2 versions. I would be open to that. What do you think?

I'm no kind of expert on WebGL but I am an expert on how MDN organizes stuff :) so I can help with that at least. I think technically `WebGL2RenderingContext` is not a subclass of `WebGLRenderingContext`: rather they both include the mixin interface `WebGLRenderingContextBase`:

```
[Exposed=(Window,Worker)]
interface WebGLRenderingContext
{
};
WebGLRenderingContext includes WebGLRenderingContextBase;
WebGLRenderingContext includes WebGLRenderingContextOverloads;
```
(https://registry.khronos.org/webgl/specs/latest/1.0/#5.14)

```
[Exposed=(Window,Worker)]
interface WebGL2RenderingContext
{
};
WebGL2RenderingContext includes WebGLRenderingContextBase;
WebGL2RenderingContext includes WebGL2RenderingContextBase;
WebGL2RenderingContext includes WebGL2RenderingContextOverloads;
```
(https://registry.khronos.org/webgl/specs/latest/2.0/#3.7)

In MDN we don't represent mixins, because we think they are a spec abstraction that developers don't need to know about. So instead we duplicate docs pages across all interfaces that include the mixin. For instance, in the Fetch API, both the `Request` and `Response` interfaces include the `Body` mixin, which defines things like `json()` and `text()` methods. But we don't explicitly document `Body` on MDN: instead we have separate pages for [`Request.json()`](https://developer.mozilla.org/en-US/docs/Web/API/Request/json) and [`Response.json()`](https://developer.mozilla.org/en-US/docs/Web/API/Response/json).

So tldr, yes, AFAICT it would be better to split https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/bufferData and https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/bufferSubData into separate pages here, and if you wanted to have a go at that I'd be happy to review it.
